metaseq/dataclass/configs.py

```diff
@@ -737,3 +748,49 @@ class MetaseqConfig(MetaseqDataclass):
     lr_scheduler: Any = MISSING
     bpe: Any = MISSING
     tokenizer: Any = None
+
+
+class DynamicConfig(dict):
```

Comment:
Should this extend `dict`? I would think that just implementing `__getitem__` might make sense, because setting items would lead to weird states.